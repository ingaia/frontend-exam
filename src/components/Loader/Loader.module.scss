@import 'scss/components.scss';

/* */

.loader {

    $size: 130px;

    position: fixed;

    top: 0;
    left: 0;

    width: 100%;
    height: 100%;

    display: flex;
    align-items: center;
    justify-content: center;

    background: #0b0b0b;

    svg {

        width: $size;
        height: $size;

        fill: $md-grey-900;

        animation: fade .3s ease;

        @keyframes fade {

            from {

                opacity: 0;
                transform: translateY(5px);

            }

            to {

                opacity: 1;
                transform: translateY(0);

            }

        }

        path {

            animation: fadeIn 1.5s ease infinite;

            animation-delay: .3s;

            @keyframes fadeIn {

                0%, 100% {

                    fill: $md-grey-900;

                }

                50% {

                    fill: $theme-color-3;

                }

            }

            &:nth-child(2){ animation-delay: .4s; }
            &:nth-child(3){ animation-delay: .7s; }

        }

    }

}
