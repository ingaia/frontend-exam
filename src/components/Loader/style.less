@import './../../variables.less';

.loader {
    border: 3px solid @hillary; 
    border-top: 3px solid transparent;
    border-radius: 50%;
    width: 76px;
    height: 76px;
    animation: spin 2s linear infinite;
    
    @-webkit-keyframes spin {
        0% { -webkit-transform: rotate(0deg); }
        100% { -webkit-transform: rotate(360deg); }
    }
      
     @keyframes spin {
        0% { transform: rotate(0deg); }
        100% { transform: rotate(360deg); }
    }
}
