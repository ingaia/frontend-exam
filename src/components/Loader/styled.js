import styled from 'styled-components';
import {colorChange} from "../../assets/animations/keyframes";

export const LoaderStyled = styled.div`
  height: 100vh;
	width: 100vw;
  background: ${props => props.theme.palette.darkLight};
  text-align: center;
`;

export const SvgStyled = styled.svg`
  fill: ${props => props.theme.palette.gray};
  width: 100px;
  padding-top: 40vh;
  
  && :first-child {
    animation: ${colorChange} 1s linear forwards;
  }
  && :nth-child(2) {
    animation: ${colorChange} 1s linear forwards;
    animation-delay: 1s;
  }
  && :last-child {
    animation: ${colorChange} 1s linear forwards;
    animation-delay: 2s;
  }
`;
