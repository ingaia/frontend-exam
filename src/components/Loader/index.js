import React from 'react';

import './style.less';

export default function Loader(){
    return (
        <div className="loader"></div>
    );
}