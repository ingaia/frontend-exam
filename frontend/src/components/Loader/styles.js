import styled from 'styled-components';

export const Container = styled.div`
    height: 100%;
    width: 100%;
    background-color: #1d1a1a;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const ToggleWrapper = styled.span`
    cursor: pointer;
    display: inline-block;
    height: 5rem;
    position: relative;
    width: 5rem;

    /* svg animations */
    @keyframes changeColor1 {
        0% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        25% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        50% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        75% {
            fill: #ff3509;
            stroke: #ff3509;
        }
        100% {
            fill: #ff3509;
            stroke: #ff3509;
        }
    }
    @keyframes changeColor2 {
        0% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        25% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        50% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        75% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        100% {
            fill: #ff3509;
            stroke: #ff3509;
        }
    }
    @keyframes changeColor3 {
        0% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        25% {
            fill: #4c4c4c;
            stroke: #4c4c4c;
        }
        50% {
            fill: #ff3509;
            stroke: #ff3509;
        }
        75% {
            fill: #ff3509;
            stroke: #ff3509;
        }
        100% {
            fill: #ff3509;
            stroke: #ff3509;
        }
    }
    .svg-1 {
        #f {
            animation: changeColor1 3s;
            animation-fill-mode: forwards;
        }
        #s {
            animation: changeColor2 3s;
            animation-fill-mode: forwards;
        }
        #t {
            animation: changeColor3 3s;
            animation-fill-mode: forwards;
        }
    }
    svg {
        height: 100%;
        overflow: visible;
        width: 100%;
        position: absolute;
        top: 0px;
        left: 0px;
    }
`;
