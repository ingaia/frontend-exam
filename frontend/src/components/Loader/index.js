import React from 'react';

import { ReactComponent as LoaderSvg } from '~/assets/loader.svg';
import { Container, ToggleWrapper } from './styles';

export default function Loader() {
    return (
        <Container>
            <ToggleWrapper>
                <LoaderSvg className="svg-1" />
            </ToggleWrapper>
        </Container>
    );
}
