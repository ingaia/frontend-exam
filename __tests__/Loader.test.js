import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Loader from '../src/components/Loader';

describe('<Loader/>', () => {
    it('should render', () => {
        const wrapper = shallow(<Loader/>);

        expect(wrapper.hasClass('loader')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});