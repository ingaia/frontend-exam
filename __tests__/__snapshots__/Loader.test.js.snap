// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> should render 1`] = `
<div
  className="loader"
/>
`;
